config/config.go

```diff
@@ -289,6 +292,12 @@ func (cfg *Config) addTransports() ([]fx.Option, error) {
 		fx.Provide(func() connmgr.ConnectionGater { return cfg.ConnectionGater }),
 		fx.Provide(func() pnet.PSK { return cfg.PSK }),
 		fx.Provide(func() network.ResourceManager { return cfg.ResourceManager }),
+		fx.Provide(func(gater connmgr.ConnectionGater, rcmgr network.ResourceManager) *tcpreuse.ConnMgr {
+			if !cfg.ShareTCPListener {
+				return nil
+			}
+			return tcpreuse.NewConnMgr(tcpreuse.EnvReuseportVal, gater, rcmgr)
```

Comment:
There seems to be a negative missing somewhere. `tcpreuse.EnvReuseportVal` is true if we should use reuseport and false otherwise, but we're passing it into a field labeled "disablereuseport". Either we should pass `!tcpreuse.EnvReuseportVal` or negate how it's used within `NewConnMgr`.

No strong feelings on which we do, although IMO passing enableX generally feels better than disableX if you are being forced to pass in a boolean.